File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n = s.size();
        vector<int>cnt(26, 0);
        for(int i = 0; i<n; i++){
            int ch = s[i]-'a';
            cnt[ch]++;
        }
        for(int i = 0; i<n; i++){
            int ch = s[i]-'a';
            if(cnt[ch]==1) return i;
        }
        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int n = s.length();
        int [] arr = new int[256];
        for(int i = 0; i<s.length(); i++){
            char ch = s.charAt(i);
            arr[ch]++;
        }
        for(int i = 0; i<s.length(); i++){
            char ch = s.charAt(i);
            if(arr[ch]==1){
                return i;
            }
        }
        return -1;
    }
}